app/Infrastructure/Controllers/GetsWalletCryptocurrenciesController.php

```diff
@@ -19,14 +19,10 @@ public function __construct(WalletDataSource $walletDataSource)
     }
     public function __invoke($wallet_id): JsonResponse
     {
-        $wallet_id = intval($wallet_id);
-        $validator = Validator::make(['wallet_id' => $wallet_id], [
-            'wallet_id' => 'required|int|min:0',
-        ]);
-        if ($validator->fails()) {
-            return response()->json([], Response::HTTP_BAD_REQUEST);
+        if (!WalletIdValidator::validateWalletId($wallet_id)) {
+            return response()->json(['error' => 'Bad Request',
+                'message' => 'The wallet id must be an integer'], Response::HTTP_BAD_REQUEST);
```

Comment:
This error is for all the cases?? And if i send a float it will be the same message?